### PR TITLE
Add carto3 examples

### PR DESCRIPTION
**Pull request by borja-munoz**
- Adapt to deck.gl 8.5.0
- Add CARTO 3 examples

In addition to this:
- Firebase deployment for Vue CARTO 2 app
- Update deck.gl version to 8.5.8 in Vue CARTO 2 app
- Remove deprecation warning for using getRadius in Vue CARTO 2 app

### CODE REVIEW

File: deck.gl/examples/pure-js/angular/carto2/package.json [hunk 1]
```diff
@@ -21,14 +21,14 @@
     "@angular/platform-browser": "~11.2.0",
     "@angular/platform-browser-dynamic": "~11.2.0",
     "@angular/router": "~11.2.0",
-    "@deck.gl/carto": "8.4.9",
-    "@deck.gl/core": "8.4.9",
-    "@deck.gl/layers": "8.4.9",
+    "@deck.gl/carto": "8.5.0",
```
Comment (VictorVelarde) [inline deck.gl/examples/pure-js/angular/carto2/package.json]: We should have now 8.6.0 right?
How about using ^8.6.0 like imports? (so the people grab this when copying examples...)

File: deck.gl/examples/pure-js/angular/carto2/src/app/modules/home/layers/rail-roads-layer.ts [hunk 2]
```diff
@@ -11,16 +11,17 @@ export class RailRoadsLayer extends Layer {
...
+      type: MAP_TYPES.QUERY,
       data: 'SELECT cartodb_id, the_geom_webmercator, category, scalerank FROM ne_10m_railroads_public',
       visible: this.visible,
       pickable: true,
       lineWidthScale: 20,
       lineWidthMinPixels: 2,
       getLineColor: colorContinuous({
         attr: 'scalerank',
-        domain: [1, 2, 3, 4, 5, 10],
+        domain: [4, 5, 6, 7, 8, 9, 10],
```
Comment (VictorVelarde) [inline deck.gl/examples/pure-js/angular/carto2/src/app/modules/home/layers/rail-roads-layer.ts:24]: ok, if you think it's better 👍🏻

File: deck.gl/examples/pure-js/angular/carto3/package.json [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    "@angular/compiler": "~11.2.0",
+    "@angular/core": "~11.2.0",
+    "@angular/forms": "~11.2.0",
+    "@angular/material": "^11.2.0",
+    "@angular/platform-browser": "~11.2.0",
+    "@angular/platform-browser-dynamic": "~11.2.0",
+    "@angular/router": "~11.2.0",
+    "@deck.gl/carto": "8.5.0",
+    "@deck.gl/core": "8.5.0",
+    "@deck.gl/layers": "8.5.0",
```
Comment (VictorVelarde) [inline deck.gl/examples/pure-js/angular/carto3/package.json]: Also ^8.6.0 here?

File: deck.gl/examples/pure-js/angular/carto3/package.json [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    "@ngrx/store": "^11.0.1",
+    "@turf/bbox-polygon": "^6.3.0",
+    "@turf/boolean-intersects": "^6.3.0",
+    "@types/mapbox-gl": "^2.1.0",
+    "deck.gl": "8.5.0",
```
Comment (VictorVelarde) [inline deck.gl/examples/pure-js/angular/carto3/package.json]: and here

File: deck.gl/examples/pure-js/vue/carto2/package.json [hunk 5]
```diff
@@ -8,11 +8,11 @@
     "lint": "vue-cli-service lint"
   },
   "dependencies": {
-    "@deck.gl/carto": "8.4.12",
+    "@deck.gl/carto": "8.5.8",
```
Comment (VictorVelarde) [inline deck.gl/examples/pure-js/vue/carto2/package.json]: ^8.6.0

File: deck.gl/examples/pure-js/vue/carto3/package.json [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  "name": "deckgl-for-vue",
+  "version": "0.1.0",
+  "private": true,
+  "scripts": {
+    "serve": "vue-cli-service serve",
+    "build": "vue-cli-service build",
+    "lint": "vue-cli-service lint"
+  },
+  "dependencies": {
+    "@deck.gl/carto": "8.5.0",
```
Comment (VictorVelarde) [inline deck.gl/examples/pure-js/vue/carto3/package.json]: ^8.6.0

File: deck.gl/examples/pure-js/vue/carto3/src/components/map-component/MapComponent.vue [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+      layerManager.init({
+        container: '#map',
+        mapStyle: this.basemap,
+        viewState: this.viewState,
+        onViewStateChange: ({ viewState }) => {
+          this[MUTATIONS.SET_VIEWSTATE](viewState)
+        },
+        // onLoad: () => {
+        //   this[MUTATIONS.SET_MAP_LOADED](true)
+        // }
```
Comment (VictorVelarde) [inline deck.gl/examples/pure-js/vue/carto3/src/components/map-component/MapComponent.vue]: remove commented, right?

**Review by VictorVelarde — COMMENTED** [review]
Overall looks good to me.
Only suggestion: upgrade to ^8.6.0. Ping me and I'll test them locally

File: deck.gl/examples/pure-js/angular/carto2/src/app/modules/home/layers/rail-roads-layer.ts (hunk 2 above)
Comment (borja-munoz) [reply deck.gl/examples/pure-js/angular/carto2/src/app/modules/home/layers/rail-roads-layer.ts:24]: Yes, there is no scalerank smaller than 4

File: deck.gl/examples/pure-js/angular/carto2/package.json (hunk 1 above)
Comment (borja-munoz) [reply deck.gl/examples/pure-js/angular/carto2/package.json]: Updated to 8.6.5. Sometimes there are breaking changes in minor versions, so I prefer to pin the example to a specific version.

File: deck.gl/examples/pure-js/angular/carto3/package.json (hunk 3 above)
Comment (borja-munoz) [reply deck.gl/examples/pure-js/angular/carto3/package.json]: Updated to 8.6.5. Sometimes there are breaking changes in minor versions, so I prefer to pin the example to a specific version.

File: deck.gl/examples/pure-js/vue/carto2/package.json (hunk 5 above)
Comment (borja-munoz) [reply deck.gl/examples/pure-js/vue/carto2/package.json]: Updated to 8.6.5. Sometimes there are breaking changes in minor versions, so I prefer to pin the example to a specific version.

File: deck.gl/examples/pure-js/vue/carto3/package.json (hunk 6 above)
Comment (borja-munoz) [reply deck.gl/examples/pure-js/vue/carto3/package.json]: Updated to 8.6.5. Sometimes there are breaking changes in minor versions, so I prefer to pin the example to a specific version.

File: deck.gl/examples/pure-js/angular/carto3/package.json (hunk 4 above)
Comment (borja-munoz) [reply deck.gl/examples/pure-js/angular/carto3/package.json]: Updated to 8.6.5. Sometimes there are breaking changes in minor versions, so I prefer to pin the example to a specific version.

File: deck.gl/examples/pure-js/vue/carto3/src/components/map-component/MapComponent.vue (hunk 7 above)
Comment (borja-munoz) [reply deck.gl/examples/pure-js/vue/carto3/src/components/map-component/MapComponent.vue]: I have uncommented the code, I don't know why I commented it out, but I've tested and it works.

**Review by VictorVelarde — APPROVED** [review]
👍🏻 . I tested both angular and vue-carto3 with latest libs, and they work well

**Pull request merged**